pennylane/labs/resource_estimation/resource_tracking.py

```diff
@@ -90,14 +90,21 @@ def resources_from_operation(
 ) -> Resources:
     """Get resources from an operation"""
 
-    if isinstance(obj, ResourceConstructor):
-        cp_rep = obj.resource_rep_from_op()
-        gate_counts_dict = defaultdict(int)
-        _counts_from_compressed_res_op(cp_rep, gate_counts_dict, gate_set=gate_set, config=config)
-        return Resources(gate_types=gate_counts_dict)
+    if not isinstance(obj, Operation):
+        raise TypeError(f"obj must be of type Operation. Got type {type(obj)}.")
```

Comment:
We don't need this since the single_dispatch won't trigger this function if the input type isn't a subclass of operation.